### PR TITLE
fix(parse): lock storage while reading using image store

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/meta/parse.go [hunk 1]
```diff
@@ -44,13 +45,19 @@ func ParseStorage(metaDB mTypes.MetaDB, storeController storage.StoreController,
...
+	imageStore.Lock(&lockLatency)
+
 	indexBlob, err := imageStore.GetIndexContent(repo)
 	if err != nil {
 		log.Error().Err(err).Str("repository", repo).Msg("load-repo: failed to read index.json for repo")
 
 		return err
 	}
 
+	imageStore.Unlock(&lockLatency)
```
Comment (andaaron) [inline pkg/meta/parse.go:59]: We should defer and unlock even in case of error.

File: pkg/meta/parse.go [hunk 2]
```diff
@@ -247,6 +254,10 @@ func getCosignSignatureLayersInfo(
 	}
 
 	for _, layer := range manifestContent.Layers {
+		var lockLatency time.Time
+
+		imageStore.Lock(&lockLatency)
```
Comment (andaaron) [inline pkg/meta/parse.go:259]: same comment as above

File: pkg/meta/parse.go (hunk 1 above)
Comment (andaaron) [reply pkg/meta/parse.go:60]: I talked with Laurentiu, and it looks like in this case we want to unlock right away, not when the function exits.

**Review by andaaron — APPROVED** [review]

File: pkg/meta/parse.go [hunk 3]
```diff
@@ -44,13 +45,20 @@ func ParseStorage(metaDB mTypes.MetaDB, storeController storage.StoreController,
 func ParseRepo(repo string, metaDB mTypes.MetaDB, storeController storage.StoreController, log log.Logger) error {
 	imageStore := storeController.GetImageStore(repo)
 
+	var lockLatency time.Time
```
Comment (peusebiu) [inline pkg/meta/parse.go:48]: why not lock the entire function? and not just parts, repo can be modified between reading its index.json here and checking its signatures leading to errors, right?

Comment (peusebiu) [reply pkg/meta/parse.go:48]: If it's running before server, and before Task scheduler, then it's fine :D

**Review by andaaron — APPROVED** [review]

**Pull request merged**